src/main/java/org/benetech/servicenet/web/rest/external/RecordsResource.java

```diff
@@ -0,0 +1,94 @@
+import org.benetech.servicenet.service.OrganizationMatchService;
+import org.benetech.servicenet.service.OrganizationService;
+import org.benetech.servicenet.service.dto.ExternalRequestRecord;
+import org.benetech.servicenet.service.dto.ExternalResponseRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+/**
+ * REST controller for managing external records request.
+ */
+@RestController
+@RequestMapping("/api")
+public class RecordsResource {
+
+    private final Logger log = LoggerFactory.getLogger(RecordsResource.class);
+
+    @Autowired
+    private OrganizationService organizationService;
+
+    @Autowired
+    private OrganizationMatchService organizationMatchService;
+
+    @Autowired
+    private ClientProfileService clientProfileService;
+
+    @PreAuthorize("hasRole('" + AuthoritiesConstants.EXTERNAL + "')")
+    @PostMapping("/records")
+    @Timed
+    public ResponseEntity<List<ExternalResponseRecord>> getOrganizationMatchInfos(
+        @Valid @RequestBody ExternalRequestRecord externalRequestRecord) throws URISyntaxException, BadRequestAlertException {
+        log.debug("REST request to get Record : {}", externalRequestRecord);
+
+        String clientId = SecurityUtils.getCurrentClientId();
+        Optional<ClientProfile> optionalClientProfile = clientProfileService.findById(clientId);
+        if (optionalClientProfile.isEmpty()) {
+            throw new BadRequestAlertException("There is provider associated with this account.",
+                RecordType.ORGANIZATION.toString(), "id");
+        }
+        ClientProfile clientProfile = optionalClientProfile.get();
+        Optional<Organization> optionalOrganization = organizationService.findByIdOrExternalDbId(externalRequestRecord.getId());
+        if (optionalOrganization.isEmpty()) {
+            throw new BadRequestAlertException("There is no organization with such id.",
+                RecordType.ORGANIZATION.toString(), "id");
+        }
+        Organization organization = optionalOrganization.get();
+        if (!clientProfile.getSystemAccount().getId().equals(organization.getAccount().getId())) {
+            throw new BadRequestAlertException("Organization not belongs to the provider.",
+                RecordType.ORGANIZATION.toString(), "id");
+        }
+        List<OrganizationMatch> matches = organizationMatchService.findAllMatchesForOrganization(organization.getId());
+
+        List<ExternalResponseRecord> results = this.mapMatchesToResponses(matches, externalRequestRecord.getSimilarity());
```

Comment:
come to think of it, we should just call this class ExternalRecordDto or put it in a separate "external" package and then it would be RecordDto